### PR TITLE
BREAKING CHANGE: update dependencies for Lit 2

### PR DESCRIPTION
**Pull request by jeroach**
This PR makes the breaking changes required to update to Lit 2 (`lit-html` 2 and `lit-element` 3). The Lit 2 upgrade will be a coordinated effort, so **this PR should not merge at this time**.

Breaking change checklist: 
- [x] Update `lit-html` -> `^2`, `lit-element` -> `^3`, `open-wc/testing` -> `^3` (if used)
- [x] `createRenderRoot` usages -> move from LitElement to ReactiveElement 
- [x] Remove Dependabot Lit rules
- [x] Bump any dependencies to their Lit 2 versions
- [x] Remove any manual lifecycle calls to Lit 1 reactive controllers
- [x] Backwards-compatible changes merged: <Your GitHub PR link>

Renamed API's:
- [x] `UpdatingElement` -> `ReactiveElement`
- [x] `@internalProperty` -> `@state` (we don't use TypeScript decorators)
- [x] `static getStyles()` -> `static finalizeStyles(styles)`
- [x] `_getUpdateComplete()` -> `getUpdateComplete()`
- [x] `NodePart` -> `ChildPart`

Testing checklist: 
- [x] CI/ unit tests pass
- [x] Local testing on demo pages
- [ ] Testing component in the LMS (required almost every other dependency ;) )

For more information on the Lit 2 upgrade, visit https://lit.dev/docs/releases/upgrade/ or contact team Gaudi.

### CODE REVIEW

**Review by amauryperez — APPROVED** [review]
Nice

File: package.json [hunk 1]
```diff
@@ -39,24 +39,23 @@
...
   "dependencies": {
-    "@brightspace-hmc/questions": "^1.0.1",
-    "@brightspace-ui-labs/facet-filter-sort": "^4.1.0",
-    "@brightspace-ui-labs/grade-result": "^1.2.5",
-    "@brightspace-ui-labs/user-profile-card": "^1",
-    "@brightspace-ui/core": "^1.220",
-    "@brightspace-ui/htmleditor": "^1",
-    "d2l-activities": "github:BrightspaceHypermediaComponents/activities#semver:^3",
-    "d2l-activity-alignments": "github:Brightspace/d2l-activity-alignments#semver:^2",
-    "d2l-html-editor": "github:Brightspace/d2l-html-editor#semver:^2",
```
Comment (jeroach) [inline package.json:50]: Removing this reference to the old `d2l-html-editor` as it doesn't seem to be used anywhere in code.

File: package.json [hunk 2]
```diff
@@ -39,24 +39,23 @@
...
+    "@brightspace-hmc/questions": "^2",
+    "@brightspace-ui-labs/facet-filter-sort": "^5",
+    "@brightspace-ui-labs/grade-result": "^2",
+    "@brightspace-ui-labs/user-profile-card": "^2",
+    "@brightspace-ui/core": "^2",
+    "@brightspace-ui/htmleditor": "^2",
+    "d2l-activities": "github:BrightspaceHypermediaComponents/activities#semver:^4",
+    "d2l-activity-alignments": "github:Brightspace/d2l-activity-alignments#semver:^3",
```
Comment (annabelsegalld2l) [inline package.json:49]: This PR should be ready to merge now with an updated package-lock file. Can you update activity-alignments to "^4" and rubric to "^6" as well? Those repos had a double-bump due to some rubrics funkiness.

Comment (jeroach) [reply package.json:49]: Sure I can bump those dependencies.

Just to confirm, I thought CE was wave 9, and according to the Slack channel, we're still on wave 3

I'll re-run the CI regardless :)

Comment (annabelsegalld2l) [reply package.json]: Yes I just changed to update each repo owner individually once we got to Phase 4 😄 Consistent eval and discover are the last repos left to migrate!

**Review by annabelsegalld2l — APPROVED** [review]

**Pull request merged**